### PR TITLE
Raise not-implemented exception on numpy fallback

### PR DESCRIPTION
**Pull request by antonwolfy**
Previously DPNP fell back on NumPy implementation when a requested functionality isn't available and can't be provided by DPNP.
The PR is indented to raise `Non-implemented` exception in that scenario by default, rather than falling back.

It will be possible to keep old behavior with falling back on NumPy calls by setting a new environment variable `DPNP_RAISE_EXCEPION_ON_NUMPY_FALLBACK = 0`.
In all other cases, the fallback is prohibited and lead to the exception.

- [x] Have you provided a meaningful PR description?
- [ ] Have you added a test, reproducer or referred to issue with a reproducer?
- [x] Have you tested your changes locally for CPU and GPU devices?
- [x] Have you made sure that new changes do not introduce compiler warnings?
- [ ] If this PR is a work in progress, are you filing the PR as a draft?

### CODE REVIEW

File: dpnp/dpnp_utils/dpnp_algo_utils.pyx [hunk 1]
```diff
@@ -114,6 +114,10 @@ def call_origin(function, *args, **kwargs):
     Call fallback function for unsupported cases
     """
 
+    if config.__DPNP_RAISE_EXCEPION_ON_NUMPY_FALLBACK__:
+        raise NotImplementedError(f"Requested funtion={function.__name__} with args={args} and kwargs={kwargs} "
+                                  "isn't currently supported, fallback on NumPy implementation.")
```
Comment (oleksandr-pavlyk) [inline dpnp/dpnp_utils/dpnp_algo_utils.pyx:119]:
```suggestion
                                  "isn't currently supported, falls back on NumPy implementation.")
```

Comment (oleksandr-pavlyk) [reply dpnp/dpnp_utils/dpnp_algo_utils.pyx:119]: or would fall back on NumPy implementation

Comment (antonwolfy) [inline dpnp/dpnp_utils/dpnp_algo_utils.pyx:119]:
```suggestion
                                  "isn't currently supported and would fall back on NumPy implementation.")
```

**Comment by oleksandr-pavlyk** [comment]
Implementation in this PR falls back by default, allows opting out.  I should clarify, but I thought we agreed to raises by default, permits opt-in.

**Comment by oleksandr-pavlyk** [comment]
> Implementation in this PR falls back by default, allows opting out. I should clarify, but I thought we agreed to raises by default, permits opt-in.

Never mind I misread the code, it works as expected

File: dpnp/random/dpnp_random_state.py [hunk 2]
```diff
@@ -125,12 +125,15 @@ def normal(self, loc=0.0, scale=1.0, size=None, dtype=None, usm_type="device"):
...
                 max_double = numpy.finfo('double').max
-                if (loc >= max_double or loc <= min_double) and dpnp.isfinite(loc):
+
+                # TODO: switch to dpnp.isfinite() and dpnp.signbit() once functions are available,
+                # but for now use direct numpy calls without call_origin() wrapper, since data is a scalar
+                if (loc >= max_double or loc <= min_double) and numpy.isfinite(loc):
                     raise OverflowError(f"Range of loc={loc} exceeds valid bounds")
 
-                if (scale >= max_double) and dpnp.isfinite(scale):
+                if (scale >= max_double) and numpy.isfinite(scale):
```
Comment (oleksandr-pavlyk) [inline dpnp/random/dpnp_random_state.py:134]: Perhaps factor out uses of `numpy.isfinite` into your own routine which can be modified as needed. 

If `scale` always remains a Python or NumPy scalar, than not compute-follows-data violation occurs, but if `scale` can be an array, then your modular routine can be modified to apply appropriate validation function.

File: tests/conftest.py [hunk 3]
```diff
@@ -73,3 +73,7 @@ def pytest_collection_modifyitems(config, items):
             # exact match of the test name with items from excluded_list
             if test_name == item_tbl_str:
                 item.add_marker(skip_mark)
+
+@pytest.fixture
+def allow_fall_back_on_numpy(monkeypatch):
+    monkeypatch.setattr(dpnp.config, '__DPNP_RAISE_EXCEPION_ON_NUMPY_FALLBACK__', 0)
```
Comment (oleksandr-pavlyk) [inline tests/conftest.py:79]: Cool trick!

File: dpnp/random/dpnp_random_state.py (hunk 2 above)
Comment (antonwolfy) [reply dpnp/random/dpnp_random_state.py]: Added internal functions _is_finite_scalar() and _is_finite_scalar() to dpnp_random_state.py with TODO comment inside to replace with dpnp functionality once availbale.

**Review by oleksandr-pavlyk — APPROVED** [review]

**Pull request merged**